src/main/java/it/pagopa/selfcare/pagopa/backoffice/service/ApiManagementService.java

```diff
@@ -61,12 +62,51 @@ public ApiManagementService(AzureApiManagerClient apimClient,
         this.environment = environment;
     }
 
-    public List<InstitutionDetail> getInstitutions() {
-        Authentication authentication = SecurityContextHolder.getContext().getAuthentication();
-        Collection<InstitutionInfo> institutions = externalApiClient.getInstitutions(Utility.extractUserIdFromAuth(authentication));
-        return institutions.stream()
-                .map(institution -> modelMapper.map(institution, InstitutionDetail.class))
-                .toList();
+    public List<InstitutionDetail> getInstitutions(String taxCode) {
+        if(taxCode != null && !taxCode.isEmpty()) {
+            return externalApiClient.getInstitutionsFiltered(taxCode).getInstitutions().stream()
+                    .map(ApiManagementService::mapInstitutionsList)
+                    .toList();
+        } else {
+            Authentication authentication = SecurityContextHolder.getContext().getAuthentication();
+            String userIdForAuth = Utility.extractUserIdFromAuth(authentication);
+            Collection<InstitutionInfo> institutions = externalApiClient.getInstitutions(userIdForAuth);
+            return institutions.stream()
+                    .map(institution -> modelMapper.map(institution, InstitutionDetail.class))
+                    .toList();
+        }
+    }
+
+    private static InstitutionDetail mapInstitutionsList(it.pagopa.selfcare.pagopa.backoffice.model.institutions.client.Institution elem) {
```

Comment:
Could be better to use model mapper or a custom converter instead of this static method for mapping the response ?